server/channels/store/sqlstore/property_field_store.go

```diff
@@ -0,0 +1,339 @@
+	}
+
+	if opts.PerPage < 0 {
+		return nil, errors.New("per page must be positive integer")
+	}
+
+	query := s.getQueryBuilder().
+		Select(propertyFieldColumns...).
+		From("PropertyFields").
+		OrderBy("CreateAt ASC").
+		Offset(uint64(opts.Page * opts.PerPage)).
+		Limit(uint64(opts.PerPage))
+
+	if !opts.IncludeDeleted {
+		query = query.Where(sq.Eq{"DeleteAt": 0})
+	}
+
+	if opts.GroupID != "" {
+		query = query.Where(sq.Eq{"GroupID": opts.GroupID})
+	}
+
+	if opts.TargetType != "" {
+		query = query.Where(sq.Eq{"TargetType": opts.TargetType})
+	}
+
+	if opts.TargetID != "" {
+		query = query.Where(sq.Eq{"TargetID": opts.TargetID})
+	}
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_search_tosql")
+	}
+
+	rows, err := s.GetReplica().Query(queryString, args...)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_search_query")
+	}
+	defer rows.Close()
+
+	fields, err := propertyFieldsFromRows(rows)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_search_propertyfieldfromrows")
+	}
+
+	return fields, nil
+}
+
+func (s *SqlPropertyFieldStore) Update(fields []*model.PropertyField) (_ []*model.PropertyField, err error) {
+	if len(fields) == 0 {
+		return nil, nil
+	}
+
+	transaction, err := s.GetMaster().Beginx()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_update_begin_transaction")
+	}
+	defer finalizeTransactionX(transaction, &err)
+
+	updateTime := model.GetMillis()
+	for _, field := range fields {
+		field.UpdateAt = updateTime
+
+		if err := field.IsValid(); err != nil {
+			return nil, errors.Wrap(err, "property_field_update_isvalid")
+		}
+
+		updateMap, err := s.propertyFieldToUpdateMap(field)
+		if err != nil {
+			return nil, err
+		}
+
+		queryString, args, err := s.getQueryBuilder().
+			Update("PropertyFields").
+			SetMap(updateMap).
+			Where(sq.Eq{"id": field.ID}).
+			ToSql()
+		if err != nil {
+			return nil, errors.Wrap(err, "property_field_update_tosql")
+		}
+
+		result, err := transaction.Exec(queryString, args...)
+		if err != nil {
+			return nil, errors.Wrap(err, "property_field_update_exec")
+		}
+
+		count, err := result.RowsAffected()
+		if err != nil {
+			return nil, errors.Wrap(err, "property_field_update_rowsaffected")
+		}
+		if count == 0 {
+			return nil, store.NewErrNotFound("PropertyField", field.ID)
+		}
+	}
+
+	if err := transaction.Commit(); err != nil {
+		return nil, errors.Wrap(err, "property_field_update_commit")
+	}
+
+	return fields, nil
+}
+
+func (s *SqlPropertyFieldStore) Delete(id string) error {
+	queryString, args, err := s.getQueryBuilder().
+		Update("PropertyFields").
+		Set("DeleteAt", model.GetMillis()).
+		Where(sq.Eq{"id": id}).
+		ToSql()
+	if err != nil {
+		return errors.Wrap(err, "property_field_delete_tosql")
+	}
+
+	result, err := s.GetMaster().Exec(queryString, args...)
+	if err != nil {
+		return errors.Wrap(err, "property_field_delete_exec")
```

Comment:
Same as earlier. Use a more user-friendly string and pass the id.

Reply:
Fixed in https://github.com/mattermost/mattermost/commit/4205b3481705802746e01f134c02ee8011d26721, PTAL